parsers/mysql/mysql.go

```diff
@@ -81,16 +100,145 @@ type SlowQuery struct {
 	LockTime        float64   `json:"lock_time"`
 	RowsSent        int       `json:"rows_sent"`
 	RowsExamined    int       `json:"rows_examined"`
-	Query           string    `json:"query",omitempty`
+	Query           string    `json:"query,omitempty"`
 	NormalizedQuery string    `json:"normalized_query,omitempty"`
+	DB              string    `json:"db,omitempty"`
+	HostedOn        string    `json:"hosted_on,omitempty"`
+	ReadOnly        bool      `json:"read_only,omitempty"`
+	ReplicaLag      int64     `json:"replica_lag,omitempty"`
+	Role            string    `json:"role,omitempty"`
 	skipQuery       bool
 }
 
-func (p *Parser) Init(_ interface{}) error {
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
 	p.nower = &RealNower{}
+	if p.conf.Host != "" {
+		url := fmt.Sprintf("%s:%s@tcp(%s)/", p.conf.User, p.conf.Pass, p.conf.Host)
+		db, err := sql.Open("mysql", url)
+		if err != nil {
+			log.Fatal(err)
+		}
+		// run one time queries
+		p.hostedOn = getHostedOn(db)
+		p.role = getRole(db)
+
+		// update hostedOn and readOnly every <n> seconds
+		go func() {
+			defer db.Close()
+			ticker := time.NewTicker(time.Second * time.Duration(p.conf.QueryInterval))
+			for _ = range ticker.C {
+				p.readOnly = getReadOnly(db)
+				p.replicaLag = getReplicaLag(db)
+			}
+		}()
+	}
 	return nil
 }
 
+func getReadOnly(db *sql.DB) bool {
+	rows, err := db.Query("SELECT @@global.read_only;")
+	if err != nil {
+		log.Println(err)
```

Comment:
with logrus, this could be, e.g. `logrus.WithError(err).Warnln("failed to get read-only state")`